cli/internal/fs/package_deps_hash.go

```diff
@@ -172,15 +173,38 @@ func UnescapeChars(in []byte) []byte {
 }
 
 // gitLsTree executes "git ls-tree" in a folder
-func gitLsTree(path string) (string, error) {
-
-	cmd := exec.Command("git", "ls-tree", "HEAD", "-r")
+func gitLsTree(path string) (map[string]string, error) {
+	cmd := exec.Command("git", "ls-tree", "HEAD", "-z", "-r")
 	cmd.Dir = path
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return "", fmt.Errorf("failed to read `git ls-tree`: %w", err)
+
+	out, pipeError := cmd.StdoutPipe()
+	if pipeError != nil {
+		return nil, fmt.Errorf("failed to read `git ls-tree`: %w", pipeError)
 	}
-	return strings.TrimSpace(string(out)), nil
+
+	startError := cmd.Start()
+	if startError != nil {
+		return nil, fmt.Errorf("failed to read `git ls-tree`: %w", startError)
+	}
+
+	reader := lstree.NewReader(out)
+	entries, readErr := reader.ReadAll()
+	if readErr != nil {
+		return nil, fmt.Errorf("failed to read `git ls-tree`: %w", readErr)
+	}
+
+	waitErr := cmd.Wait()
```

Comment:
Do we need the split between `Start` and `Wait`? Can we just use `Run` instead? Or does that mess up the streaming somehow?

Reply:
I sourced the ordering from [the `Cmd.StdoutPipe` example](https://pkg.go.dev/os/exec#example-Cmd.StdoutPipe). I believe it matters; I hypothesize if the `Wait()` call comes first that the command has to exit before getting piped to the `Reader`.